astropy/units/physical.py

```diff
@@ -1,133 +1,425 @@
+        consistent with the physical type of the `PhysicalType` instance.
+        """
+        if isinstance(other, (str, set)):
+            other = self._standardize_physical_type_names(other)
+
+        if isinstance(other, set):
+            return other.issubset(self.as_set)
+        elif isinstance(other, PhysicalType):
+            return self._get_physical_type_id() == other._get_physical_type_id()
+        else:
+            return False
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __contains__(self, item):
+        return item in self.as_set
+
+    def __repr__(self):
+        return repr(self.as_string) if len(self.as_set) == 1 else self.as_string
+
+    def __str__(self):
+        return self.as_string
+
+    @staticmethod
+    def _identify_unit_from_unit_or_physical_type(obj) -> core.Unit:
+        """
+        If a unit is passed in, return that unit.  If a physical type is
+        passed in, return a unit that corresponds to that physical type.
+        """
+        if isinstance(obj, core.UnitBase):
+            return obj
+        elif isinstance(obj, PhysicalType):
+            return obj._unit
+        else:
+            raise TypeError("Expecting a unit or a physical type")
+
+    def __mul__(self, other):
+        other_unit = self._identify_unit_from_unit_or_physical_type(other)
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = self._unit * other_unit
+        return new_unit.physical_type
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        other_unit = self._identify_unit_from_unit_or_physical_type(other)
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = self._unit / other_unit
+        return new_unit.physical_type
+
+    def __rtruediv__(self, other):
+        other_unit = self._identify_unit_from_unit_or_physical_type(other)
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = other_unit / self._unit
+        return new_unit.physical_type
+
+    def __pow__(self, power):
+        if not isinstance(power, numbers.Real):
+            raise TypeError(f"{power} is not a real number")
+        return (self._unit ** power).physical_type
+
+
+def def_physical_type(unit: core.UnitBase, name: Union[str, Set[str]]):
+    """
+    Add a mapping between a unit and the corresponding physical type(s).
+
+    Parameters
+    ----------
+    unit : u.Unit
+        The unit to be represented by the physical type.
+
+    name : `str` or `set` of `str`
+        A `str` representing the name of the physical type of the unit,
+        or a `set` containing strings that represent one or more names
+        of physical types.
+    """
+
+    if name in ("unknown", {"unknown"}) or "unknown" in name:
+        raise ValueError("Unable to uniquely define an unknown physical type")
```

Comment:
Maybe try to cover? Also total nitpick: python style seems to be to start exception messages with lower case (since they follow a colon when written out)